web/Controllers/AdminController.cs

```diff
@@ -26,4 +29,40 @@ public IActionResult Index()
 
         return View(viewModel);
     }
+
+    [AllowAnonymous]
+    [HttpPost]
+    public IActionResult RequestAdminAccess()
+    {
+        var isAuthenticated = User.Identity?.IsAuthenticated ?? false;
+
+        if (!isAuthenticated)
+        {
+            _logger.LogWarning("Not authenticated user");
+            TempData["FlashType"] = "danger";
+            TempData["FlashMessage"] = "Not logged in";
+
+            return RedirectToAction(nameof(Index));
+        }
+
+        var user = GetCurrentUser();
+        if (user.IsAdmin)
+        {
+            _logger.LogWarning("User {UserId} already an admin",user.Id);
+            TempData["FlashType"] = "danger";
+            TempData["FlashMessage"] = "Already an admin";
+
+            return RedirectToAction(nameof(Index));
+        }
+
+        _adminService.RequestAdminAccess(user);

```

Comment:
Is there anything that could go wrong in this method? Any errors to catch?

Reply:
FWIW I would vote not to have the extra error handling for the user not being found. It's not a clear-cut thing, but I think a generic error page is reasonable - currently it should be impossible, and even if we added a way to delete users that happening between the start of the request and here (we check the user against the database on every request currently) is unlikely and would probably mean there's a bug rather than the user doing something wrong.

Of course things could change and the method might be called in a way that makes it more likely, but I'd generally prefer to keep things simple.

Anyway feel free to keep the code or revert it, fine by me either way.

(it's fine to answer "No" or start a discussion to review comments from us if you don't think a suggestion makes sense, and especially when we're asking a question)

Reply:
Ah you beat me to it Ben 😄 I was just going to say, this isn't quite what I had intended. I was thinking more along the lines of "if `GetCurrentUser` returns `null`, do something" but I'm not even sure if that method can return `null`.

Like Ben says, when I comment a question it's not a trick, I'm genuinely starting a discussion 😄 I'd probably revert the code to the way it was, as it was clearer to see what was going on. 😊

Reply:
It's been reverted! Sorry about that 😅